functions/__fzf_search_current_dir.fish

```diff
@@ -5,20 +5,20 @@ function __fzf_search_current_dir --description "Search the current directory. R
 
     set fd_opts --color=always $fzf_fd_opts
     set fzf_arguments --multi --ansi
-    set current_token (commandline --current-token)
-    set token (string unescape -- $current_token)
+    set token (commandline --current-token)
     # need to expand ~ in the directory name since fd can't expand it
     set expanded_token (string replace --regex -- "^~/" $HOME/ $token)
+    set unescaped_token (string unescape -- $expanded_token)
 
     # If the current token is a directory and has a trailing slash,
     # then use it as fd's base directory.
-    if string match --quiet -- "*/" $token && test -d $expanded_token
-        set --append fd_opts --base-directory=$expanded_token
+    if string match --quiet -- "*/" $token && test -d "$unescaped_token"
+        set --append fd_opts --base-directory=$unescaped_token
         # use the directory name as fzf's prompt to indicate the search is limited to that directory
-        set --append fzf_arguments --prompt=$token --preview="__fzf_preview_file $token{}"
-        set file_paths_selected $expanded_token(fd $fd_opts 2>/dev/null | fzf $fzf_arguments)
+        set --append fzf_arguments --prompt="$unescaped_token" --preview="__fzf_preview_file $expanded_token{}"
```

Comment:
Here's bugfix 2: we need to pass the escaped (but expanded) token to __fzf_preview_file in order for the preview to work for paths containing spaces.